src/services/timeout.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * This file is part of Threema Web.
+ *
+ * Threema Web is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Threema Web. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export class TimeoutService {
+    private logTag: string = '[TimeoutService]';
+
+    // Config
+    private config: threema.Config;
+
+    // Angular services
+    private $log: ng.ILogService;
+    private $timeout: ng.ITimeoutService;
+
+    // List of registered timeouts
+    private timeouts: Set<ng.IPromise<any>> = new Set();
+
+    public static $inject = ['CONFIG', '$log', '$timeout'];
+    constructor(config: threema.Config, $log: ng.ILogService, $timeout: ng.ITimeoutService) {
+        this.config = config;
+        this.$log = $log;
+        this.$timeout = $timeout;
+    }
+
+    /**
+     * Log a message on debug log level, but only if the `DEBUG` flag is enabled.
+     */
+    private logDebug(msg: string): void {
+        if (this.config.DEBUG) {
```

Comment:
I would drop that for now. This will be resolved by adding the micrologger from SaltyRTC.

Reply:
The timeout logs are really noisy (dozens of logs when scrolling through the list of contacts), therefore I wouldn't do that.

Reply:
Then please add a `TODO` note so we don't forget about it.

Reply:
If we add a new logging wrapper, we'll have to touch every line of code that does logging anyways. The DEBUG condition is hard to overlooks :slightly_smiling_face: